src/transformers/models/greaselm/processing_greaselm.py

```diff
@@ -0,0 +1,267 @@
+        kg_encoding = self.feature_extractor(question_answer_example, entailed_qa, num_choices)
+
+        # Step 4: combine the two encodings ready for model input
+        return LMKGEncoding(data={**lm_encoding, **kg_encoding})
+
+    @staticmethod
+    def detect_type(question_answer_example: Dict[str, Any]):
+        if "question" in question_answer_example and "question_concept" in question_answer_example["question"]:
+            return "commonsenseqa"
+        elif "question" in question_answer_example and "stem" in question_answer_example["question"]:
+            # most likely openbookqa
+            return "openbookqa"
+        else:
+            return None
+
+    def encode_question_answer_example(self, entailed_qa_examples: List[Dict[str, Any]]):
+        class InputExample(object):
+            def __init__(self, example_id, question, contexts, endings, label=None):
+                self.example_id = example_id
+                self.question = question
+                self.contexts = contexts
+                self.endings = endings
+                self.label = label
+
+        def read_examples(qa_entailed_statements: List[Dict[str, Any]]) -> List[InputExample]:
+            examples = []
+            for json_dic in qa_entailed_statements:
+                label = ord(json_dic["answerKey"]) - ord("A") if "answerKey" in json_dic else 0
+                contexts = json_dic["question"]["stem"]
+                if "para" in json_dic:
+                    contexts = json_dic["para"] + " " + contexts
+                if "fact1" in json_dic:
+                    contexts = json_dic["fact1"] + " " + contexts
+                examples.append(
+                    InputExample(
+                        example_id=json_dic["id"],
+                        contexts=[contexts] * len(json_dic["question"]["choices"]),
+                        question="",
+                        endings=[ending["text"] for ending in json_dic["question"]["choices"]],
+                        label=label,
+                    )
+                )
+
+            return examples
+
+        def examples_to_features(examples: List[InputExample], label_list: List[int]) -> List[Dict[str, Any]]:
+            label_map = {label: i for i, label in enumerate(label_list)}
+
+            features = []
+            for idx, example in tqdm(enumerate(examples), total=len(examples)):
+                choices = []
+                for context, ending in zip(example.contexts, example.endings):
+                    ans = example.question + " " + ending
+                    choices.append((context, ans))
+                encoded_input = self.tokenizer(
+                    choices,
+                    padding="max_length",
+                    truncation=True,
+                    max_length=self.max_seq_length,
+                    return_token_type_ids=True,
+                    return_tensors="pt",
+                    return_special_tokens_mask=True,
+                )
```

Comment:
In other processors, one can typically provide kwargs themselves, and then the processor will forward those to its tokenizer (like padding, truncation, whether to add special tokens or not, etc.).

This seems to be all fixed here. Could we provide this flexibility to the processor?

Reply:
Great catch. I am afraid to do it and then mess up something :-) Let's leave it as is now and then if a need arises we can adjust.